### PR TITLE
Fix native image warning in Kafka

### LINKED ISSUE #32461: Quarkus 3.0.0.CR2 - quarkus-kafka-client triggers NativeImageBuildStep in JVM mode
### Describe the bug

Quarkus 3.0.0.CR2 - `quarkus-kafka-client` extension triggers NativeImageBuildStep in JVM mode

Regression comparing to 3.0.0.CR1, visible through this warning on LINUX machines

```
 [WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Cannot find the `native-image` in the 
  GRAALVM_HOME, JAVA_HOME and System PATH. Install it using `gu install native-image` 
  Attempting to fall back to container build.
```

Other extensions do not trigger this warnings. 

### Expected behavior

No warning from NativeImageBuildStep

### Actual behavior
```
 [WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Cannot find the `native-image` in the 
  GRAALVM_HOME, JAVA_HOME and System PATH. Install it using `gu install native-image` 
  Attempting to fall back to container build.
```

### How to Reproduce?

generate sample app using `mvn io.quarkus.platform:quarkus-maven-plugin:3.0.0.CR2:create -DprojectGroupId=my-groupId -DprojectArtifactId=app-quarkus-kafka-client -DprojectVersion=1.0.0-SNAPSHOT -DpackageName=org.my.group -DquarkusRegistryClient=false -DplatformVersion=3.0.0.CR2 -Dextensions=quarkus-kafka-client`

run `mvn clean package -DskipTests -f app-quarkus-kafka-client`

Check the log for the warning

CR1 has no warning on Linux machines using `mvn clean package -DskipTests -f app-quarkus-kafka-client -Dquarkus.platform.version=3.0.0.CR1` command

### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

Java 11

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.0.CR2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: https://github.com/quarkusio/quarkus/issues/32461

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
LGTM but let's wait for @yrodiere's approval too.

**Review by yrodiere — APPROVED** [review]
LGTM, thanks @geoand!

**Comment by geoand** [comment]
YW!

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java [hunk 1]
```diff
@@ -286,13 +289,11 @@ public void build(
...
         handleOpenTracing(reflectiveClass, capabilities);
         handleStrimziOAuth(curateOutcomeBuildItem, reflectiveClass);
-        if (config.snappyEnabled) {
-            handleSnappy(nativeImageRunner, reflectiveClass, nativeLibs);
-        }
 
     }
 
-    private void handleSnappy(NativeImageRunnerBuildItem nativeImageRunner,
+    @BuildStep(onlyIf = { IsSnappy.class, NativeOrNativeSourcesBuild.class })
```
Comment (cescoffier) [inline extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:295]: Nice catch!

Comment (geoand) [reply extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:295]: 🙏🏼

**Pull request merged**